src/lib.rs

```diff
@@ -343,53 +344,76 @@ mod serde;
 #[cfg(feature = "kv_unstable")]
 pub mod kv;
 
-#[cfg(has_atomics)]
-use std::sync::atomic::{AtomicUsize, Ordering};
-
-#[cfg(not(has_atomics))]
-use std::cell::Cell;
-#[cfg(not(has_atomics))]
-use std::sync::atomic::Ordering;
-
-#[cfg(not(has_atomics))]
-struct AtomicUsize {
-    v: Cell<usize>,
-}
-
-#[cfg(not(has_atomics))]
-impl AtomicUsize {
-    const fn new(v: usize) -> AtomicUsize {
-        AtomicUsize { v: Cell::new(v) }
-    }
+// In case it has atomics. See https://github.com/rust-lang/log/issues/489.
+#[cfg(not(all(
+    target = "thumbv4t-none-eabi",
+    target = "msp430-none-elf",
+    target = "riscv32i-unknown-none-elf",
+    target = "riscv32imc-unknown-none-elf",
+)))]
+use std::sync::atomic::AtomicUsize;
+
+// In case it doesn't have atomics. See https://github.com/rust-lang/log/issues/489.
+#[cfg(any(
+    target = "thumbv4t-none-eabi",
+    target = "msp430-none-elf",
+    target = "riscv32i-unknown-none-elf",
+    target = "riscv32imc-unknown-none-elf",
+))]
+mod not_atomics {
+    use super::Ordering;
+    use std::cell::Cell;
+
+    struct AtomicUsize {
+        v: Cell<usize>,
+    }
+
+    impl AtomicUsize {
+        const fn new(v: usize) -> AtomicUsize {
+            AtomicUsize { v: Cell::new(v) }
+        }
 
-    fn load(&self, _order: Ordering) -> usize {
-        self.v.get()
-    }
+        fn load(&self, _order: Ordering) -> usize {
+            self.v.get()
+        }
 
-    fn store(&self, val: usize, _order: Ordering) {
-        self.v.set(val)
-    }
+        fn store(&self, val: usize, _order: Ordering) {
+            self.v.set(val)
+        }
 
-    #[cfg(atomic_cas)]
-    fn compare_exchange(
-        &self,
-        current: usize,
-        new: usize,
-        _success: Ordering,
-        _failure: Ordering,
-    ) -> Result<usize, usize> {
-        let prev = self.v.get();
-        if current == prev {
-            self.v.set(new);
+        // In case it has atomic_cas.
+        #[cfg(any(
```

Comment:
This duplicates the outer module's cfg

Reply:
No it doesn't, the difference is the first one:

`thumbv4t-none-eabi` vs `thumbv6m-none-eabi`

I didn't want to only keep this one so it would make it easier for potential future updates.